### DISCUSSION TITLE
[Question] Rollbacking transactions with Promise.all()

### DISCUSSION THREAD

**Discussion by rubenlopezlozoya** [mikro-orm/mikro-orm#3165] [Q&A]
Hey team, we are trying to run a POC using Mikro ORM and one of the pain points that we are trying to solve coming from our existing implementation using other ORM is handling rollbacks of transactions when using Promise.all() inside. Right now, we found out that in other ORMS, when we do something like this:

```
try {
  prisma.$transaction((prisma) => async {
   const updates = items.map((item) => {
      prisma.updateMany(....)
   });
   await Promise.all(updates);
  })
} catch (err) {
  // Handle the rollback...
}
```

if any of the promises rejects, the transaction will rollback but only all the operations that happened before the failing one. The rest of the operations that are being executed as part of the Promise.all() will still reach the database, leaving us with an undesired partial update. I would expect ORMs to handle this situation for me and somehow disregard/rollback all ongoing operations after one of the promises inside a transaction fails. I am wondering if what I want to achieve can be done using mikro ORM out of the box or if I will have to play around with Promise.allSettled API (or just going full sequential) to get what I want.

Sorry if this is not the right forum to ask this question, but I went through the docs and despite reading about explicity transactions I am not sure if this will be handled by Mikro ORM.

**Comment by B4nan** [comment]
If you ask me, you should leverage the UoW and automatic batch updates instead of using `Promise.all`. For a single flush call you will get a single update query per table, which is pretty much always more performant than an update query per row. Just update the entities and call flush at the end, UoW will handle the rest for you.

Other than that, I guess `Promise.allSettled` should do what you want. Not sure what playing do you mean, its just about replacing single word and things should work, isn't it? :]
